.buildkite/create_integration_testing_worksheet.py

```diff
@@ -6,61 +6,106 @@
+
 
-def copy_worksheet():
-    google_account = gspread.authorize(ServiceAccountCredentials.from_json_keyfile_name(SPREADSHEET_CREDENTIALS, SCOPE))
-    wks_open = google_account.open_by_key(SPREADSHEET_FILE_KEY)
-    wks_id = wks_open.get_worksheet(SHEET_TEMPLATE_INDEX).id
-    return wks_open.duplicate_sheet(wks_id, insert_sheet_index=1, new_sheet_id=None, new_sheet_name=SHEET_TAG)
+def create_spreadsheet():
+    return GOOGLE_ACCOUNT.copy(SPREADSHEET_TEMPLATE_KEY, title=SPREADSHEET_TITLE, copy_permissions=True)
 
 
-def get_worksheet_link(wks_id):
+def get_worksheet_link(sheet_id, wks_id):
     wks_domain = "https://docs.google.com/spreadsheets/d/"
     link_arrg = "{0}{1}{2}{3}"
-    return link_arrg.format(wks_domain, SPREADSHEET_FILE_KEY, "/edit#gid=", wks_id)
-
-
-def edit_new_worksheet():
-    worksheet = copy_worksheet()
-    print(worksheet.id)
-    count_cell_empty = 0
-    for count in itertools.count():
-        current_cell = SHEET_TEMPLATE_FEATURES_COLUMN + str(count + 1)
-        cell_value = worksheet.acell(current_cell, value_render_option='FORMULA').value
-        edited_cell_value = cell_value.replace(SHEET_TEMPLATE_TAG, SHEET_TAG)
-        if SHEET_TAG in edited_cell_value:
-            count_cell_empty = 0
-            worksheet.update_acell(current_cell, edited_cell_value)
-        else:
-            if not str(edited_cell_value):
-                count_cell_empty += 1
-            if count_cell_empty >= SHEET_MAX_EMPTY_CELL:
-                break
-    wks_link = get_worksheet_link(worksheet.id)
-    print("Here's the new integration testing worksheet link =====>", wks_link)
+    return link_arrg.format(wks_domain, sheet_id, "/edit#gid=", wks_id)
+
+
+def get_feature_name(str_arg):
+    str_name = str_arg.replace("-", " ").replace(".feature", " ").capitalize()
+    return str_name.strip()
+
+
+def get_role_name(str_arg):
+    return str_arg.replace("-", " ").capitalize() + " scenarios"
+
+
+def get_feature_dir_path():
+    patent_path = os.path.dirname(os.getcwd())
+    return patent_path + "/integration_testing/features/"
+
+
+def check_file_exist(str_arg):
+    return os.path.exists(str_arg)
+
+
+def check_name(str_arg1, str_arg2):
+    if str_arg2 == "":
+        return False
+    elif check_file_exist(str_arg1) and str_arg2[0] != "#":
+        return True
+    return False
+
+
+def fetch_feature_files():
+
+    feature_dir = get_feature_dir_path()
+    spreadsheet = create_spreadsheet()
+    worksheet = spreadsheet.get_worksheet(SHEET_INDEX)
+    order_name = "/ORDER.txt"
+    role_order_path = feature_dir + order_name
+    counter = 3
+    with open(role_order_path, 'r') as read_role:
+        role_lines = read_role.readlines()
+        for role in role_lines:
+            role_file_name = role.strip()
+            role_path = feature_dir + role_file_name + "/"
+            role_order_path = role_path + order_name
```

Comment:
Please avoid re-using the same variable/s (example: `role_order_path`) inside inner code blocks to prevent confusion.